lib/console/deployments/observability/webhook/datadog.ex

```diff
@@ -0,0 +1,180 @@
+    case find_tag(tags, ["plrl_service", "service", "plural_service"]) do
+      {_, val} -> Map.put(acc, :service_id, service(id, val))
+      _ -> acc
+    end
+  end
+
+  def associations(:service, %{"meta" => %{"service" => service}}, %{cluster_id: id} = acc) when is_binary(service) and is_binary(id),
+    do: Map.put(acc, :service_id, service(id, service))
+
+  def associations(:service, %{"service" => service}, %{cluster_id: id} = acc) when is_binary(service) and is_binary(id),
+    do: Map.put(acc, :service_id, service(id, service))
+
+  def associations(scope, payload, acc) when is_binary(payload) do
+    extract_from_text(scope, payload, acc)
+  end
+
+  def associations(scope, payload, acc) when is_map(payload) do
+    text = extract_text_from_payload(payload)
+    extract_from_text(scope, text, acc)
+  end
+
+  def associations(_, _, acc), do: acc
+
+  def state("triggered"), do: :firing
+  def state("alert"), do: :firing
+  def state("warning"), do: :firing
+  def state("error"), do: :firing
+  def state("critical"), do: :firing
+  def state("alerting"), do: :firing
+  def state("trigger"), do: :firing
+  def state("firing"), do: :firing
+
+  def state("ok"), do: :resolved
+  def state("resolved"), do: :resolved
+  def state("recovery"), do: :resolved
+  def state("recover"), do: :resolved
+  def state("normal"), do: :resolved
+  def state(_), do: :resolved
+
+  def severity(%{"priority" => "P1"}), do: :critical
+  def severity(%{"priority" => "P2"}), do: :high
+  def severity(%{"priority" => "P3"}), do: :medium
+  def severity(%{"priority" => "P4"}), do: :low
+  def severity(%{"priority" => "P5"}), do: :low
+
+  def severity(%{"alert_type" => "error"}), do: :critical
+  def severity(%{"alert_type" => "critical"}), do: :critical
+  def severity(%{"alert_type" => "warning"}), do: :medium
+  def severity(%{"alert_type" => "info"}), do: :low
+  def severity(_), do: :undefined
+
+  def summary(%{"title" => title, "body" => body}) when is_binary(title) and is_binary(body),
+    do: "#{title}\n#{body}\n"
+  def summary(%{"text_title" => title}) when is_binary(title),
+    do: "Alert Summary: #{title}\n"
+  def summary(%{"message" => message}) when is_binary(message),
+    do: "Alert Summary: #{message}\n"
+  def summary(%{"body" => body}) when is_binary(body),
+    do: "Alert Summary: #{body}\n"
+  def summary(_), do: ""
+
+  # Helper functions
+
+  defp find_tag(tags, possible_keys) do
+    Enum.find_value(tags, fn
+      {k, v} -> if k in possible_keys, do: {k, v}, else: nil
+      tag when is_binary(tag) ->
+        case String.split(tag, ":", parts: 2) do
+          [k, v] -> if k in possible_keys, do: {k, v}, else: nil
+          _ -> nil
+        end
+      _ -> nil
+    end)
+  end
+
+  defp extract_from_text(:project, text, acc) do
+    case extract_with_regex(text, @project_regex) do
+      nil -> acc
+      val -> Map.put(acc, :project_id, project(val))
+    end
+  end
+
+  defp extract_from_text(:cluster, text, acc) do
```

Comment:
all of this is already done by the raw extractor that you likely are using in the webhook module referencing this